lib/ansible/modules/cloud/vmware/vmware_host_powerstate.py

```diff
@@ -0,0 +1,222 @@
+  delegate_to: localhost
+  register: reboot_host
+
+'''
+
+RETURN = r'''
+result:
+    description: metadata about host system's state
+    returned: always
+    type: dict
+    sample: {
+        "esxi01": {
+            "msg": "power down 'esxi01' to standby",
+            "error": "",
+        },
+    }
+'''
+
+try:
+    from pyVmomi import vim, vmodl
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import PyVmomi, vmware_argument_spec, wait_for_task, TaskError
+from ansible.module_utils._text import to_native
+
+
+class VmwareHostPowerManager(PyVmomi):
+    def __init__(self, module):
+        super(VmwareHostPowerManager, self).__init__(module)
+        cluster_name = self.params.get('cluster_name')
+        esxi_host_name = self.params.get('esxi_hostname')
+        self.hosts = self.get_all_host_objs(cluster_name=cluster_name, esxi_host_name=esxi_host_name)
+        if not self.hosts:
+            self.module.fail_json(msg="Failed to find host system with given configuration.")
+
+    def ensure(self):
+        """
+        Function to manage internal state of host system
+
+        """
+        results = dict(changed=False, result=dict())
+        state = self.params.get('state')
+        force = self.params.get('force')
+        timeout = self.params.get('timeout')
+        host_change_list = []
+        for host in self.hosts:
+            changed = False
+            if not host.runtime.inMaintenanceMode and not force:
+                self.module.fail_json(msg="Current host system '%s' is not in maintenance mode,"
+                                          " please specify 'force' as True to proceed." % host.name)
+            if host.runtime.connectionState == 'notResponding':
+                self.module.fail_json(msg="Current host system '%s' can not be set in '%s'"
+                                          " mode as the host system is not responding." % (host.name, state))
+
+            results['result'][host.name] = dict(msg='', error='')
+            if state == 'reboot-host' and not host.capability.rebootSupported:
+                self.module.fail_json(msg="Current host '%s' can be rebooted as the host system"
+                                          " does not have capability to reboot." % host.name)
+            elif state == 'shutdown-host' and not host.capability.shutdownSupported:
+                self.module.fail_json(msg="Current host '%s' can be shut down as the host system"
+                                          " does not have capability to shut down." % host.name)
+            elif state in ['power-down-to-standby', 'power-up-from-standby'] and not host.capability.standbySupported:
+                self.module.fail_json(msg="Current host '%s' can be %s as the host system"
+                                          " does not have capability to standby supported." % host.name)
+
+            verb = "reboot '%s'" % host.name
```

Comment:
shouldn't this be in the `if state == 'reboot-host':` part? Potentially make `verb = ""` and see if it set before you wait for the task and mark as changed?